src/main/java/frc/robot/commands/DynamicDefaultDriveCommand.java

```diff
@@ -0,0 +1,119 @@
+      DoubleSupplier translationYSupplier,
+      DoubleSupplier rotationSupplier) {
+    this.drivetrainSubsystem = drivetrainSubsystem;
+    this.robotAngleSupplier = robotAngleSupplier;
+    this.translationXSupplier = translationXSupplier;
+    this.translationYSupplier = translationYSupplier;
+    this.rotationSupplier = rotationSupplier;
+
+    this.useSlewRate = true;
+    addRequirements(drivetrainSubsystem);
+  }
+
+  public DynamicDefaultDriveCommand(
+      DrivetrainSubsystem drivetrainSubsystem,
+      Supplier<Rotation2d> robotAngleSupplier,
+      DoubleSupplier translationXSupplier,
+      DoubleSupplier translationYSupplier,
+      DoubleSupplier rotationSupplier,
+      boolean useSlewRate) {
+    this.drivetrainSubsystem = drivetrainSubsystem;
+    this.robotAngleSupplier = robotAngleSupplier;
+    this.translationXSupplier = translationXSupplier;
+    this.translationYSupplier = translationYSupplier;
+    this.rotationSupplier = rotationSupplier;
+
+    this.useSlewRate = useSlewRate;
+    addRequirements(drivetrainSubsystem);
+  }
+
+  @Override
+  public void initialize() {
+    var robotAngle = robotAngleSupplier.get();
+
+    // Calculate field relative speeds
+    var chassisSpeeds = drivetrainSubsystem.getChassisSpeeds();
+    var robotSpeeds =
+        new ChassisSpeeds(
+            chassisSpeeds.vxMetersPerSecond * robotAngle.getCos()
+                - chassisSpeeds.vyMetersPerSecond * robotAngle.getSin(),
+            chassisSpeeds.vyMetersPerSecond * robotAngle.getCos()
+                + chassisSpeeds.vxMetersPerSecond * robotAngle.getSin(),
+            chassisSpeeds.omegaRadiansPerSecond);
+
+    // Reset the slew rate limiters, in case the robot is already moving
+    translateXRateLimiter.reset(robotSpeeds.vxMetersPerSecond);
+    translateYRateLimiter.reset(robotSpeeds.vyMetersPerSecond);
+    rotationRateLimiter.reset(robotSpeeds.omegaRadiansPerSecond);
+  }
+
+  @Override
+  public void execute() {
+    // You can use `new ChassisSpeeds(...)` for robot-oriented movement instead of field-oriented
+    // movement
+    if (useSlewRate) {
+      drivetrainSubsystem.drive(
+          ChassisSpeeds.fromFieldRelativeSpeeds(
+              translateXRateLimiter.calculate(translationXSupplier.getAsDouble()),
+              translateYRateLimiter.calculate(translationYSupplier.getAsDouble()),
+              -rotationRateLimiter.calculate(rotationSupplier.getAsDouble()),
+              robotAngleSupplier.get()));
+    } else {
+      drivetrainSubsystem.drive(
+          ChassisSpeeds.fromFieldRelativeSpeeds(
+              translationXSupplier.getAsDouble(),
+              translationYSupplier.getAsDouble(),
+              -rotationSupplier.getAsDouble(),
+              robotAngleSupplier.get()));
+    }
+  }
+
+  @Override
+  public void end(boolean interrupted) {
+    drivetrainSubsystem.stop();
+  }
+
+  public void toggleSlewRate() {
+    this.useSlewRate = !this.useSlewRate;
```

Comment:
When you enable slew rate limiter, you will want to re-initialize it to the current velocities